File: casadaApi/src/main/java/com/casada/casadaApi/entities/Buyer.java
package com.casada.casadaApi.entities;

import lombok.Data;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.io.Serializable;

/**
 * $table.getTableComment()
 */
@Data
@Entity
@Table(name = "buyer")
public class Buyer implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "buyerid", nullable = false)
    private Long buyerid;

    @Column(name = "buyername", nullable = false)
    private String buyername;

    @Column(name = "buyersurname", nullable = false)
    private String buyersurname;

    @Column(name = "buyerphonenumber", nullable = false)
    private String buyerphonenumber;

    @Column(name = "buyerhomeaddress", nullable = false)
    private String buyerhomeaddress;

    @Column(name = "buyerdeliveryaddress")
    private String buyerdeliveryaddress;

    @Column(name = "buyeremail")
    private String buyeremail;

    @Column(name = "homepostalcode", nullable = false)
    private Long homepostalcode;

    @Column(name = "deliverypostalcode")
    private Long deliverypostalcode;

}

File: casadaGUI/casada/lib/data/buyer.dart
class Buyer {
  int? buyerId;
  String? buyerName;
  String? buyerSurname;
  String? buyerPhoneNumber;
  String? buyerHomeAddress;
  String? buyerDeliveryAddress;
  String? buyerEmail;
  int? homePostalCode;
  String? homePostalCodeName;
  int? deliveryPostalCode;
  String? deliveryPostalCodeName;

  Buyer(
      {this.buyerId,
      this.buyerName,
      this.buyerSurname,
      this.buyerPhoneNumber,
      this.buyerHomeAddress,
      this.buyerDeliveryAddress,
      this.buyerEmail,
      this.homePostalCode,
      this.homePostalCodeName,
      this.deliveryPostalCode,
      this.deliveryPostalCodeName});

  Buyer.fromJson(Map<String, dynamic> json) {
    buyerId = json['buyerId'];
    buyerName = json['buyerName'];
    buyerSurname = json['buyerSurname'];
    buyerPhoneNumber = json['buyerPhoneNumber'];
    buyerHomeAddress = json['buyerHomeAddress'];
    buyerDeliveryAddress = json['buyerDeliveryAddress'];
    buyerEmail = json['buyerEmail'];
    homePostalCode = json['homePostalCode'];
    homePostalCodeName = json['homePostalCodeName'];
    deliveryPostalCode = json['deliveryPostalCode'];
    deliveryPostalCodeName = json['deliveryPostalCodeName'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['buyerId'] = this.buyerId;
    data['buyerName'] = this.buyerName;
    data['buyerSurname'] = this.buyerSurname;
    data['buyerPhoneNumber'] = this.buyerPhoneNumber;
    data['buyerHomeAddress'] = this.buyerHomeAddress;
    data['buyerDeliveryAddress'] = this.buyerDeliveryAddress;
    data['buyerEmail'] = this.buyerEmail;
    data['homePostalCode'] = this.homePostalCode;
    data['homePostalCodeName'] = this.homePostalCodeName;
    data['deliveryPostalCode'] = this.deliveryPostalCode;
    data['deliveryPostalCodeName'] = this.deliveryPostalCodeName;
    return data;
  }

  Buyer copyWith({
    int? buyerId,
    String? buyerName,
    String? buyerSurname,
    String? buyerPhoneNumber,
    String? buyerHomeAddress,
    String? buyerDeliveryAddress,
    String? buyerEmail,
    int? homePostalCode,
    String? homePostalCodeName,
    int? deliveryPostalCode,
    String? deliveryPostalCodeName,
  }) {
    return Buyer(
      buyerId: buyerId ?? this.buyerId,
      buyerName: buyerName ?? this.buyerName,
      buyerSurname: buyerSurname ?? this.buyerSurname,
      buyerPhoneNumber: buyerPhoneNumber ?? this.buyerPhoneNumber,
      buyerHomeAddress: buyerHomeAddress ?? this.buyerHomeAddress,
      buyerDeliveryAddress: buyerDeliveryAddress ?? this.buyerDeliveryAddress,
      buyerEmail: buyerEmail ?? this.buyerEmail,
      homePostalCode: homePostalCode ?? this.homePostalCode,
      homePostalCodeName: homePostalCodeName ?? this.homePostalCodeName,
      deliveryPostalCode: deliveryPostalCode ?? this.deliveryPostalCode,
      deliveryPostalCodeName: deliveryPostalCodeName ?? this.deliveryPostalCodeName,
    );
  }
}
